### DISCUSSION TITLE
trying to run sbt (scala code) via dotnet/aspire

### DISCUSSION THREAD

**Discussion by schmitch** [microsoft/aspire#6672] [Q&A] [answered]
Hello I have a scala application that I want to run with `dotnet/aspire` it's a playframework application, however when I try to run it it crashes the aspire dashboard and it does not work:

```
var builder = DistributedApplication.CreateBuilder(args);

var baseDirectory = Directory.GetParent(builder.AppHostDirectory)?.FullName;
ArgumentException.ThrowIfNullOrEmpty(baseDirectory);

Console.WriteLine($"Using {baseDirectory} as a base path");

var isWin = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
var isMac = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);

builder
    .AddExecutable("scala-web", "sbt", baseDirectory,
        args: ["run", "-Dplay.server.http.port=9001", "-Dsbt.server.autostart=false"]);

builder.Build().Run();
```
```
<Project Sdk="Microsoft.NET.Sdk">
  
  <Sdk Name="Aspire.AppHost.Sdk" Version="9.0.0" />

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <IsAspireHost>true</IsAspireHost>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Aspire.Hosting.AppHost" Version="9.0.0" />
    <PackageReference Include="Aspire.Hosting.NodeJs" Version="9.0.0" />
  </ItemGroup>
</Project>
```

What might be problematic is that sbt starts in a reloading mode, i.e. something like `dotnet watch`. After the run I press Ctrl+C however aspire dashboard keeps running besides that it does not work anymore and I cant rerun it.
the dashboard can be accessed shortly after starting and it also prints that it starts playframework however it does not print everything and than the log just hangs and nothing works.

if I use:

```
var process = new Process
{
    StartInfo = new ProcessStartInfo
    {
        WorkingDirectory = baseDirectory,
        FileName = "sbt",
        UseShellExecute = true,
        Arguments = "run -Dplay.server.http.port=9001 -Dsbt.server.autostart=false",
    }
};
process.Start();
```
basically everthing works.

Is there something I'm missing?

**Comment by davidfowl** [comment]
1. What do you mean "it crashes", the dashboard crashes or the play application crashes?
2. You're saying it crashes but then you mention that pressing ctrl+c keeps the dashboard running.


Can you record a video and post it? Its not super clear from your description what you're seeing in the dashboard and the app host?

**Reply by schmitch** [reply]
1. it does not "crash" but it does not respond
2. it keeps the port open, I can record it

https://github.com/user-attachments/assets/d8cd5ac1-0857-4939-a2c6-479a3724c4cf

so basically now I need to manually kill all these processes, since if I restart aspire it will says that the port 17058 is still in use.
aspire does not respond, however it's somehow alive? the webserver of the play application also does not respond, but it opens the webserver port as well?
if I'm fast enough the aspire dashboard, will show the logs that the application starts and prints the normal logs of the application until a certain point.

**Reply by davidfowl** [reply]
From the video it looks like it does shutdown, nothing crashes, but there's some delay in the shut down. Is that right? Or are you seeing the processes never die no matter how long you wait?

**Reply by schmitch** [reply]
the process does never die and as said neither the dashboard nor the application is accessible. it just hangs forever in the browser, no matter if I wait with ctrl+c or not. 
I can even curl which will just timeout.

if I comment out the sbt executable part and add a simple dotnet app, everything works.

**Reply by davidfowl** [reply, marked as answer]
I created an issue https://github.com/dotnet/aspire/issues/6715
